### PR TITLE
Provider/Azure: Update vmss and vmss vm client.

### PR DESCRIPTION
**Pull request by nilo19**
Update vmss and vmss vm client to the new version in cloud provider azure.

Partly solves #2898.

/area provider/azure

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/azure_client.go [hunk 1]
```diff
@@ -23,25 +23,20 @@ import (
...
-// VirtualMachineScaleSetsClient defines needed functions for azure compute.VirtualMachineScaleSetsClient.
-type VirtualMachineScaleSetsClient interface {
-	Get(ctx context.Context, resourceGroupName string, vmScaleSetName string) (result compute.VirtualMachineScaleSet, err error)
-	CreateOrUpdate(ctx context.Context, resourceGroupName string, name string, parameters compute.VirtualMachineScaleSet) (resp *http.Response, err error)
-	DeleteInstances(ctx context.Context, resourceGroupName string, vmScaleSetName string, vmInstanceIDs compute.VirtualMachineScaleSetVMInstanceRequiredIDs) (resp *http.Response, err error)
-	List(ctx context.Context, resourceGroupName string) (result []compute.VirtualMachineScaleSet, err error)
-}
-
 // VirtualMachineScaleSetVMsClient defines needed functions for azure compute.VirtualMachineScaleSetVMsClient.
 type VirtualMachineScaleSetVMsClient interface {
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_client.go]: remove VirtualMachineScaleSetVMsClient ?

File: cluster-autoscaler/cloudprovider/azure/azure_manager_test.go [hunk 2]
```diff
@@ -57,10 +67,110 @@ func TestCreateAzureManagerValidConfig(t *testing.T) {
...
 		MaxDeploymentsCount: 8,
+		CloudProviderRateLimitConfig: providerazure.CloudProviderRateLimitConfig{
+			RateLimitConfig: azclients.RateLimitConfig{
+				CloudProviderRateLimit:            false,
+				CloudProviderRateLimitBucket:      5,
+				CloudProviderRateLimitBucketWrite: 5,
+				CloudProviderRateLimitQPS:         1,
+				CloudProviderRateLimitQPSWrite:    1,
+			},
+			RouteRateLimit: &azclients.RateLimitConfig{
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_manager_test.go]: let's remove the unused rate limit config here, e.g. route, LB, subnet, interface and so on

**Comment by nilo19** [comment]
> could you update docs on the ratelimit and backoff retry options?

Updated.

**Comment by dharmab** [comment]
Really happy to see this PR, it will help with API usage.

Is this a candidate for backport to previous minor versions, or only for the latest minor version?

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 3]
```diff
@@ -78,9 +93,25 @@ type AzureManager struct {
...
+// CloudProviderRateLimitConfig indicates the rate limit config for each clients.
+type CloudProviderRateLimitConfig struct {
+	// The default rate limit config options.
+	azclients.RateLimitConfig
+
+	// Rate limit config for each clients. Values would override default settings above.
+	InterfaceRateLimit              *azclients.RateLimitConfig `json:"interfaceRateLimit,omitempty" yaml:"interfaceRateLimit,omitempty"`
+	VirtualMachineRateLimit         *azclients.RateLimitConfig `json:"virtualMachineRateLimit,omitempty" yaml:"virtualMachineRateLimit,omitempty"`
+	StorageAccountRateLimit         *azclients.RateLimitConfig `json:"storageAccountRateLimit,omitempty" yaml:"storageAccountRateLimit,omitempty"`
+	DiskRateLimit                   *azclients.RateLimitConfig `json:"diskRateLimit,omitempty" yaml:"diskRateLimit,omitempty"`
```
Comment (marwanad) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go]: is this needed? Similar question for StorageAccount and VirtualMachines?

Comment (nilo19) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: I deleted the unneeded and keep vmss and vm. I think vm is needed, I'll update vm client in another pr.

Comment (feiskyer) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: let's only keep rate limit configs for the clients existing in azure_client.go

**Pull request merged**